server/src/main/java/org/opensearch/plugins/SecureAuxTransportSettingsProvider.java

```diff
@@ -34,9 +36,10 @@ default Optional<SSLContext> buildSecureAuxServerTransportContext(Settings setti
 
     /**
      * Additional params required for configuring ALPN.
+     * @param transport the auxiliary transport to be provided SecureAuxTransportParameters.
      * @return an instance of {@link SecureAuxTransportSettingsProvider.SecureAuxTransportParameters}
      */
-    default Optional<SecureAuxTransportSettingsProvider.SecureAuxTransportParameters> parameters() {
+    default Optional<SecureAuxTransportSettingsProvider.SecureAuxTransportParameters> parameters(AuxTransport transport) {
```

Comment:
I believe all our  `parameters` are using `Settings settings`, by inspecting the AUX transport from settings, the provider has all the info which ones are needed:

```suggestion
    default Optional<SecureAuxTransportSettingsProvider.SecureAuxTransportParameters> parameters(Settings settings) {
```

Reply:
>  think I still need the AuxTransport reference (or at least a key by which I can identify the transport) as these security settings are defined inside the security plugin

This could be done using `AUX_TRANSPORT_TYPES_KEY`, the problem with `AuxTransport` is that whatever the implementation is, the security plugin will not have access to the type (hopefully), so the only use would be rudimentary class name check. We could take it one step further:

```
default Optional<SecureAuxTransportSettingsProvider.SecureAuxTransportParameters> parameters(Settings settings, String type) {
  ....
}
```

So to identify the AUX transport type in question

Reply:
Moved these to `Settings settings, String auxTransportSettingKey` to avoid exposing the transport reference to plugins.